JAMES-2315 Split community from contribute

chibenwa opened the pull request:
The rationals: Make clear that **GitHub is a contribution mean**, and not intended to distribute software.

Please note that solving downloads will be done in a separate PR.

I took the chance to improve the contributing guidelines we desperately needed to improve...

Amongst them:
 - The split allows to better present how to contribute.
 - Contribution part is intended to be more guided and welcoming
 - To move contribution as a TLD concern on the main webpage
 - Details more the way to contribute
 - Clearly expose git[hub] is for contributing and not a way to ship the product

Hope you will enjoy.

The community page becomes:

![capture d ecran de 2018-01-31 17-21-11](https://user-images.githubusercontent.com/6928740/35617578-2efad510-06ab-11e8-955e-d02a9d29aee6.png)

The new contribute section looks like:

![capture d ecran de 2018-01-31 17-44-58](https://user-images.githubusercontent.com/6928740/35618760-7e9f3108-06ae-11e8-8a36-977f4fe047ba.png)


This work is based on top of: https://github.com/linagora/james-project/pull/1279

rouazana commented:
Your screenshot seems to not match your code. For example I don't see the words "in order of increasing level" (and that's pretty a good thing).

chibenwa commented:
I will regenerate them.

rouazana commented on src/site/xdoc/contribute.xml:
```diff
@@ -81,112 +93,93 @@
+      <p>
+        Clone the source code of the project from its 
+        <a href="https://github.com/apache/james-project">Github</a>
+        or its <a href="git://git.apache.org/james-project.git">apache git repository</a>
```
please inverse the 2 links

rouazana commented on src/site/xdoc/contribute.xml:
```diff
@@ -205,58 +198,46 @@
+        be GREATLY appreciated as this will help to improve the James project.
       </p>
+      <p>6. The standard Apache license header <strong>MUST</strong> be placed at the top of every file.</p>
+      <p>7. Your change set <strong>MUST</strong> be covered by tests. Extra bonus points for integration tests.</p>
```
s/Extra bonus points/We also strongly appreciate

chibenwa commented:
> "Report a bug" has to be in Contribute section

@mbaechler  that is funny because that was initially what I tended to do, before I changed my mind. I think the difference is tiny, and JIRA might deserve to be in contribute as well as in community. Hence I added a "report bugs" line in contribute.

The thing is, JIRA can get the same usages than emails, to report  a problem, share a feature, a fix, and thus IMO deserved to be presented close to it. (And thus in the "community" section).

What made change my mind was that I wanted "contribute" and "community" to be the same size.

Anyway, I concider your remark 100% valid and applied it in another commit. If every one is happy with the lastest version then that's perfect!

![capture d ecran de 2018-02-01 09-37-05](https://user-images.githubusercontent.com/6928740/35658364-84ecff9c-0733-11e8-9ae1-217013d92286.png)

![capture d ecran de 2018-02-01 09-37-18](https://user-images.githubusercontent.com/6928740/35658374-8cf4e222-0733-11e8-917e-f643996b5990.png)

chibenwa commented:
Merged
